### PR TITLE
feat(statsd): Emit tokio runtime metrics via statsd

### PR DESCRIPTION
**Pull request by Dav1dde**
#skip-changelog

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[build]
+rustflags = ["--cfg", "tokio_unstable"]
```
Comment (jjbayer) [inline .cargo/config.toml:2]: I hope this means "API unstable", not "unreliable"?

File: relay-server/src/services/stats.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        metric!(gauge(TokioGauges::BudgetForcedYieldCount) = m.budget_forced_yield_count());
+        metric!(gauge(TokioGauges::NumBlockingThreads) = m.num_blocking_threads() as u64);
+        metric!(gauge(TokioGauges::NumIdleBlockingThreads) = m.num_idle_blocking_threads() as u64);
+        metric!(gauge(TokioGauges::NumWorkers) = m.num_workers() as u64);
+        for worker in 0..m.num_workers() {
+            let worker_name = worker.to_string();
+            metric!(
+                gauge(TokioGauges::WorkerLocalQueueDepth) =
+                    m.worker_local_queue_depth(worker) as u64,
+                worker = &worker_name,
```
Comment (jjbayer) [inline relay-server/src/services/stats.rs:41]: Are we OK with the cardinality this produces (`num_pods * num_workers`)?

File: relay-server/src/services/stats.rs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            }
+        }
+    }
+}
+
+impl Service for RelayStats {
+    type Interface = ();
+
+    fn spawn_handler(self, _rx: relay_system::Receiver<Self::Interface>) {
+        let mut ticker = interval(Duration::from_secs(5));
```
Comment (jjbayer) [inline relay-server/src/services/stats.rs]: I feel like we should make this configurable from the start, if only to have a killswitch to disable it (i.e. interpret `0` as disabled).

File: .cargo/config.toml (hunk 1 above)
Comment (Dav1dde) [reply .cargo/config.toml:2]: That's what it means (API unstable). A bunch of it is already stable but not all of it.

File: relay-server/src/services/stats.rs (hunk 2 above)
Comment (Dav1dde) [reply relay-server/src/services/stats.rs:41]: What's the alternative? Can we emit a custom gauge with min/max/avg?

Comment (iker-barriocanal) [reply relay-server/src/services/stats.rs:41]: The number of workers is static, so I think it should be fine.

File: relay-server/src/statsd.rs [hunk 4]
```diff
@@ -48,6 +50,95 @@ impl GaugeMetric for RelayGauges {
...
+    /// Exposes [`RuntimeMetrics::num_blocking_threads`].
+    NumBlockingThreads,
+    /// Exposes [`RuntimeMetrics::num_idle_blocking_threads`].
+    NumIdleBlockingThreads,
+    /// Exposes [`RuntimeMetrics::num_workers`].
+    NumWorkers,
+    /// Exposes [`RuntimeMetrics::worker_local_queue_depth`].
+    ///
+    /// This metric is tagged with:
+    /// - `worker`: the worker id
```
Comment (iker-barriocanal) [inline relay-server/src/statsd.rs]: nit (same with below):
```suggestion
    /// - `worker`: the worker id.
```

**Review by iker-barriocanal — APPROVED** [review]
Are we trying to debug an issue with these metrics? Could we elaborate the PR description?

**Review by jjbayer — APPROVED** [review]
Looking forward to having these metrics!

Please make the interval configurable (and killswitchable), rest LGTM.

**Pull request merged**